### ISSUE TITLE
Rounding error in unit conversion

### ISSUE THREAD

**Issue by drpeterfoster** [hgrecco/pint#1028]
It seems that the unit registry conversion is introducing precision errors.
```
>>> import pint
>>> ur = pint.UnitRegistry()
>>> ur.Quantity(1, 'uM').m_as('nM')
999.9999999999999

>>> ur.Quantity(1, 'nM').m_as('uM')
0.0010000000000000002
```
pint=0.10.1
python=3.6

**Comment by hgrecco** [comment]
This is indeed a rounding error due to floating point numbers. We now provide a basic support for other numerical types. On 0.12 (March 2020) we will likely merge #1000 providing full support for other numerical types and this will no longer be an issue.

**Comment by hgrecco** [comment]
Now that #1000 has been merged this is getting better:
```python
>>> ureg.Quantity(Decimal(1), 'uM').m_as('nM') # this is correct
Decimal('1000')
>>> ureg.Quantity(Decimal(1), 'nM').m_as('uM') # this is not correct yet
Decimal('0.001000')
```

**Comment by hgrecco** [comment]
You can also use Fraction type:
```python
>>> from fractions import Fraction
>>> ureg = pint.UnitRegistry(non_int_type=Fraction)
>>> ureg.parse_expression('1 uM').m_as('nM')
Fraction(1000, 1)
>>> ureg.parse_expression('1 nM').m_as('uM')
Fraction(1, 1000)
```

**Comment by hgrecco** [comment]
I am creating a new issue to discuss what is pending after #1000
